### PR TITLE
[PS-2425] Upgrade vulnerable packages

### PR DESCRIPTION
**Pull request by goremykin**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
My ide said that this solution contains some [vulnerable packages](https://devhub.checkmarx.com/cve-details/CVE-2022-30187/). I didn't figure out if this attack is difficult, I just decided to update.

## Code changes

* **Core.csproj:** Upgrade vulnerable packages
* **bitwarden-server.sln:** Remove non-existent files from solution files
* **StripeSubscriptions.cshtml:** Remove duplicated attribute

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: bitwarden-server.sln [hunk 1]
```diff
@@ -13,10 +13,8 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 	ProjectSection(SolutionItems) = preProject
 		.editorconfig = .editorconfig
 		.gitignore = .gitignore
-		build.sh = build.sh
 		.github\workflows\build.yml = .github\workflows\build.yml
 		Directory.Build.props = Directory.Build.props
-		dotnet-tools.json = dotnet-tools.json
 		LICENSE.txt = LICENSE.txt
```
Comment (MGibson1) [inline bitwarden-server.sln:18]: Why were these lines removed?

File: src/Core/Core.csproj [hunk 2]
```diff
@@ -24,8 +24,8 @@
     <PackageReference Include="AWSSDK.SimpleEmail" Version="3.7.0.150" />
     <PackageReference Include="AWSSDK.SQS" Version="3.7.2.47" />
     <PackageReference Include="Azure.Extensions.AspNetCore.DataProtection.Blobs" Version="1.2.1" />
-    <PackageReference Include="Azure.Storage.Blobs" Version="12.11.0" />
-    <PackageReference Include="Azure.Storage.Queues" Version="12.9.0" />
+    <PackageReference Include="Azure.Storage.Blobs" Version="12.14.1" />
+    <PackageReference Include="Azure.Storage.Queues" Version="12.12.0" />
```
Comment (MGibson1) [inline src/Core/Core.csproj:28]: There should be `package.lock.json` updates to go along with these changes. To generate them you can either

1. Revert these changes and update the package by running `nuget add package Azure.Storage.Blobs` (and again for Queues)
2. Delete the `obj` directory found at `src/Core/obj` and running `dotnet restore`

**Comment by MGibson1** [comment]
Thank you for the contribution and pointing this out! I appreciate the help notifying. It does look like there are a few items to clean up prior to merging, though.

Let me know if you're able to update this PR. If not, I'm happy to push this along.

File: src/Admin/Views/Tools/StripeSubscriptions.cshtml [hunk 3]
```diff
@@ -125,7 +125,7 @@
                 <button type="submit" class="btn btn-primary" title="Search" name="action" asp-for="Action" value="@StripeSubscriptionsAction.Search"><i class="fa fa-search"></i> Search</button>
         </div>
     <hr/>
-    <input type="checkbox" class="d-none" asp-for="Filter.SelectAll" name="filter.SelectAll" id="selectAllInput" asp-for="Model.Filter.SelectAll">
+    <input type="checkbox" class="d-none" asp-for="Filter.SelectAll" name="filter.SelectAll" id="selectAllInput">
```
[the remark this answers is not in the document]
Comment (goremykin) [reply src/Admin/Views/Tools/StripeSubscriptions.cshtml:128]: @MGibson1, this is intended change :)
The `asp-for` attribute us duplicated that is why I decided to remove one

File: bitwarden-server.sln (hunk 1 above)
Comment (goremykin) [reply bitwarden-server.sln:18]: @MGibson1, This section represents "Solution items". I noticed that some files no longer exist
image

File: src/Core/Core.csproj (hunk 2 above)
Comment (goremykin) [reply src/Core/Core.csproj:28]: @MGibson1, Oh, now I see. Will do this today. Thanks for the review!

**Comment by goremykin** [comment]
@MGibson1, Done 🙂

File: src/Admin/Views/Tools/StripeSubscriptions.cshtml (hunk 3 above)
[the remark this answers is not in the document]
Comment (MGibson1) [reply src/Admin/Views/Tools/StripeSubscriptions.cshtml:128]: Ah, I see that now; thank you for pointing it out.

However, it's totally unrelated to the dependency update so I'd still like to separate these changes into a more atomic set of PRs.

File: bitwarden-server.sln (hunk 1 above)
Comment (MGibson1) [reply bitwarden-server.sln:18]: Hmm. It seems these files have been moved over the years.

Each project has a build.sh and dotnet-tools.json has been moved to `.config/dotnet-tools.json`

However, like the other change, I'd prefer to keep the dependency update atomic, can you revert this and we can figure out what to do about this in a separate PR?

**Review by MGibson1 — CHANGES_REQUESTED** [review]
Dependency update looks good, now.

Only change requested is to split the PR into the three distinct parts. This one should focus just on the dep change.

**Comment by goremykin** [comment]
@MGibson1, Done :slightly_smiling_face:

**Review by MGibson1 — APPROVED** [review]
Yay thank you :D

**Pull request merged**